### PR TITLE
make MSI package vendor information configurable

### PR DESCRIPTION
**Pull request by junyuan-z**
change all `.wxl` files extension to `.wxl.template`
update batch script to replace `{}` text template in `.wxl.template` files

sample usage:

1. `VENDOR`: `Microsoft`
2. `VENDOR_BRANDING`: `Microsoft Build of OpenJDK`

### CODE REVIEW

File: wix/Includes/OpenJDK.Variables.wxi.template [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
   <?define ProductVersion="$(var.ProductMajorVersion).$(var.ProductMinorVersion).$(var.ProductMaintenanceVersion).$(var.ProductPatchVersion)" ?>
   <?define ProductLanguage="!(loc.ProductLanguage)" ?>
-  <?define ProductManufacturer="AdoptOpenJDK" ?>
+  <?define ProductManufacturer="{vendor}OpenJDK" ?>
```
Comment (aahlenst) [inline wix/Includes/OpenJDK.Variables.wxi.template:16]: Vendor variable should replace the entire string. "DragonwellOpenJDK" or "CorrettoOpenJDK" would be both wrong and bad.

File: wix/Lang/OpenJDK.Base.de-de.wxl.template [hunk 2]
```diff
@@ -13,10 +13,10 @@
...
 	<String Id="FeatureJarFileRunWithTitle">Associate .jar</String>
-	<String Id="FeatureJarFileRunWithDescription">Associate .jar files to run with AdoptOpenJDK</String>
+	<String Id="FeatureJarFileRunWithDescription">Associate .jar files to run with {branding}</String>
 	<String Id="DowngradeDenied">"Eine neuere Version von [ProductName] ist bereits installiert. Das Setup wird jetzt beendet."</String>
 	<String Id="FeatureOracleJavaSoftName">JavaSoft (Oracle) registry keys</String>
-	<String Id="FeatureOracleJavaSoftDescription">Overwrites the reg keys HKLM\Software\JavaSoft (Oracle). After uninstallation of AdoptOpenJDK, Oracle Java running from PATH "C:\Program Files (x86)\Common Files\Oracle\Java\javapath" does not work. You need to reinstall it to recreate these registry keys</String>
+	<String Id="FeatureOracleJavaSoftDescription">Overwrites the reg keys HKLM\Software\JavaSoft (Oracle). After uninstallation of {branding}, Oracle Java running from PATH "C:\Program Files (x86)\Common Files\Oracle\Java\javapath" does not work. You need to reinstall it to recreate these registry keys</String>
```
Comment (aahlenst) [inline wix/Lang/OpenJDK.Base.de-de.wxl.template:19]: Dot missing at the end of the last sentence.

File: wix/Lang/OpenJDK.Base.en-us.wxl.template [hunk 3]
```diff
@@ -14,10 +14,10 @@
...
 	<String Id="FeatureJarFileRunWithTitle">Associate .jar</String>
-	<String Id="FeatureJarFileRunWithDescription">Associate .jar files to run with AdoptOpenJDK</String>
+	<String Id="FeatureJarFileRunWithDescription">Associate .jar files to run with {branding}</String>
 	<String Id="DowngradeDenied">"A later version of [ProductName] is already installed. Setup will now exit."</String>
 	<String Id="FeatureOracleJavaSoftName">JavaSoft (Oracle) registry keys</String>
-	<String Id="FeatureOracleJavaSoftDescription">Overwrites the reg keys HKLM\Software\JavaSoft (Oracle). After uninstallation of AdoptOpenJDK, Oracle Java running from PATH "C:\Program Files (x86)\Common Files\Oracle\Java\javapath" does not work. You need to reinstall it to recreate these registry keys</String>
+	<String Id="FeatureOracleJavaSoftDescription">Overwrites the reg keys HKLM\Software\JavaSoft (Oracle). After uninstallation of {branding}, Oracle Java running from PATH "C:\Program Files (x86)\Common Files\Oracle\Java\javapath" does not work. You need to reinstall it to recreate these registry keys</String>
```
Comment (aahlenst) [inline wix/Lang/OpenJDK.Base.en-us.wxl.template:20]: Dot missing at the end of final sentence.

File: wix/Main.hotspot.wxs.template [hunk 4]
```diff
@@ -124,10 +124,10 @@
         HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\FileExts\.jar\OpenWithProgids
         is automatically created by Windows when running jar file for the first time
         -->
-        <RegistryValue Root="HKMU" Key="SOFTWARE\Classes\.jar" Type="string" Value="AdoptOpenJDK.jarfile" KeyPath="yes" />
+        <RegistryValue Root="HKMU" Key="SOFTWARE\Classes\.jar" Type="string" Value="{vendor}OpenJDK.jarfile" KeyPath="yes" />
```
Comment (aahlenst) [inline wix/Main.hotspot.wxs.template:127]: Remove "OpenJDK"

File: wix/Main.hotspot.wxs.template [hunk 5]
```diff
@@ -124,10 +124,10 @@
...
         <RegistryValue Root="HKMU" Key="SOFTWARE\Classes\.jar" Type="string" Name="Content Type" Value="application/java-archive" KeyPath="no" />
 
-        <RegistryValue Root="HKMU" Key="SOFTWARE\Classes\AdoptOpenJDK.jarfile\shell\open\command" Type="string" Value="&quot;[INSTALLDIR]bin\javaw.exe&quot; -jar &quot;%1&quot; %*" KeyPath="no" />
+        <RegistryValue Root="HKMU" Key="SOFTWARE\Classes\{vendor}OpenJDK.jarfile\shell\open\command" Type="string" Value="&quot;[INSTALLDIR]bin\javaw.exe&quot; -jar &quot;%1&quot; %*" KeyPath="no" />
```
Comment (aahlenst) [inline wix/Main.hotspot.wxs.template:130]: Remove "OpenJDK"

File: wix/Main.hotspot.wxs.template [hunk 6]
```diff
@@ -154,10 +154,10 @@
             HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\FileExts\.jnlp\OpenWithProgids
             is automatically created by Windows when running jnlp file for the first time
             -->
-            <RegistryValue Root="HKMU" Key="SOFTWARE\Classes\.jnlp" Type="string" Value="AdoptOpenJDK.jnlpfile" KeyPath="yes" />
+            <RegistryValue Root="HKMU" Key="SOFTWARE\Classes\.jnlp" Type="string" Value="{vendor}OpenJDK.jnlpfile" KeyPath="yes" />
```
Comment (aahlenst) [inline wix/Main.hotspot.wxs.template:157]: Remove "OpenJDK"

File: wix/Main.hotspot.wxs.template [hunk 7]
```diff
@@ -154,10 +154,10 @@
...
             <RegistryValue Root="HKMU" Key="SOFTWARE\Classes\.jnlp" Type="string" Name="Content Type" Value="application/jnlp" KeyPath="no" />
 
-            <RegistryValue Root="HKMU" Key="SOFTWARE\Classes\AdoptOpenJDK.jnlpfile\shell\open\command" Type="string" Value="&quot;[INSTALLDIR]bin\javaws.exe&quot; -jnlp &quot;%1&quot; %*" KeyPath="no" />
+            <RegistryValue Root="HKMU" Key="SOFTWARE\Classes\{vendor}OpenJDK.jnlpfile\shell\open\command" Type="string" Value="&quot;[INSTALLDIR]bin\javaws.exe&quot; -jnlp &quot;%1&quot; %*" KeyPath="no" />
```
Comment (aahlenst) [inline wix/Main.hotspot.wxs.template:160]: Remove "OpenJDK"

File: wix/Main.openj9.wxs.template [hunk 8]
```diff
@@ -124,10 +124,10 @@
         HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\FileExts\.jar\OpenWithProgids
         is automatically created by Windows when running jar file for the first time
         -->
-        <RegistryValue Root="HKMU" Key="SOFTWARE\Classes\.jar" Type="string" Value="AdoptOpenJDK.jarfile" KeyPath="yes" />
+        <RegistryValue Root="HKMU" Key="SOFTWARE\Classes\.jar" Type="string" Value="{vendor}OpenJDK.jarfile" KeyPath="yes" />
```
Comment (aahlenst) [inline wix/Main.openj9.wxs.template:127]: Remove "OpenJDK"

File: wix/Main.openj9.wxs.template [hunk 9]
```diff
@@ -124,10 +124,10 @@
...
         <RegistryValue Root="HKMU" Key="SOFTWARE\Classes\.jar" Type="string" Name="Content Type" Value="application/java-archive" KeyPath="no" />
 
-        <RegistryValue Root="HKMU" Key="SOFTWARE\Classes\AdoptOpenJDK.jarfile\shell\open\command" Type="string" Value="&quot;[INSTALLDIR]bin\javaw.exe&quot; -jar &quot;%1&quot; %*" KeyPath="no" />
+        <RegistryValue Root="HKMU" Key="SOFTWARE\Classes\{vendor}OpenJDK.jarfile\shell\open\command" Type="string" Value="&quot;[INSTALLDIR]bin\javaw.exe&quot; -jar &quot;%1&quot; %*" KeyPath="no" />
```
Comment (aahlenst) [inline wix/Main.openj9.wxs.template:130]: Remove "OpenJDK"

File: wix/Main.openj9.wxs.template [hunk 10]
```diff
@@ -154,10 +154,10 @@
             HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\FileExts\.jnlp\OpenWithProgids
             is automatically created by Windows when running jnlp file for the first time
             -->
-            <RegistryValue Root="HKMU" Key="SOFTWARE\Classes\.jnlp" Type="string" Value="AdoptOpenJDK.jnlpfile" KeyPath="yes" />
+            <RegistryValue Root="HKMU" Key="SOFTWARE\Classes\.jnlp" Type="string" Value="{vendor}OpenJDK.jnlpfile" KeyPath="yes" />
             <RegistryValue Root="HKMU" Key="SOFTWARE\Classes\.jnlp" Type="string" Name="Content Type" Value="application/jnlp" KeyPath="no" />
 
-            <RegistryValue Root="HKMU" Key="SOFTWARE\Classes\AdoptOpenJDK.jnlpfile\shell\open\command" Type="string" Value="&quot;[INSTALLDIR]bin\javaws.exe&quot; -jnlp &quot;%1&quot; %*" KeyPath="no" />
+            <RegistryValue Root="HKMU" Key="SOFTWARE\Classes\{vendor}OpenJDK.jnlpfile\shell\open\command" Type="string" Value="&quot;[INSTALLDIR]bin\javaws.exe&quot; -jnlp &quot;%1&quot; %*" KeyPath="no" />
```
Comment (aahlenst) [inline wix/Main.openj9.wxs.template:160]: Remove "OpenJDK"

File: wix/Main.openj9.wxs.template [hunk 11]
```diff
@@ -154,10 +154,10 @@
             HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\FileExts\.jnlp\OpenWithProgids
             is automatically created by Windows when running jnlp file for the first time
             -->
-            <RegistryValue Root="HKMU" Key="SOFTWARE\Classes\.jnlp" Type="string" Value="AdoptOpenJDK.jnlpfile" KeyPath="yes" />
+            <RegistryValue Root="HKMU" Key="SOFTWARE\Classes\.jnlp" Type="string" Value="{vendor}OpenJDK.jnlpfile" KeyPath="yes" />
```
Comment (aahlenst) [inline wix/Main.openj9.wxs.template:157]: Remove "OpenJDK"

File: wix/Lang/OpenJDK.hotspot.de-de.wxl.template [hunk 12]
```diff
@@ -0,0 +1,12 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (douph1) [inline wix/Lang/OpenJDK.hotspot.de-de.wxl.template:1]: This file and others below where deleted and added, not moved by git
Don't we lose git history ?

**Review by douph1 — CHANGES_REQUESTED** [review]
Please don't delete CreateSourceFolder.OpenJDK11.ps1 and CreateSourceFolder.OpenJDK8.ps1
there are used to build many installer at once to test changes

File: wix/Lang/OpenJDK.hotspot.de-de.wxl.template (hunk 12 above)
Comment (junyzheng) [reply wix/Lang/OpenJDK.hotspot.de-de.wxl.template:1]: I am using `git mv` but I am not sure why those files show up as deleted and added...

Comment (junyzheng) [reply wix/Lang/OpenJDK.hotspot.de-de.wxl.template:1]: this commit `https://github.com/AdoptOpenJDK/openjdk-installer/pull/252/commits/d657ca2aaa5391c5606476cc06d481f748230741` is rename, this commit is `https://github.com/AdoptOpenJDK/openjdk-installer/pull/252/commits/80a6eab8df56d6518835ecdda3177d79b3e061fd` update the file. but putting together make the file look like deleted and added :(

Comment (junyzheng) [reply wix/Lang/OpenJDK.hotspot.de-de.wxl.template]: I am running into the issue this guy had before: https://dev.to/nickytonline/renaming-a-file-via-git-after-it-has-been-deleted-in-a-pr-41dn

Comment (junyzheng) [reply wix/Lang/OpenJDK.hotspot.de-de.wxl.template:1]: do you want me to create 2 PRs? the first one renaming files, and the second one adds parameters and updates the cmd file.

Comment (douph1) [reply wix/Lang/OpenJDK.hotspot.de-de.wxl.template:1]: For the rename / move don't worry, it's all good.
git log --follow wix/Lang/OpenJDK.Base.de-de.wxl.template : OK
git blame : OK
and 
git log --name-status
show rename with 77% similarity
https://stackoverflow.com/questions/53056942/git-diff-name-status-what-does-r100-mean

>commit 5e46b26fb91e39736050ef702f833fc84da98128
>Author: Junyuan Zheng <junyuanz@microsoft.com>
>Date:   Wed Oct 7 10:31:34 2020 -0700
>
>    rename Lang/*
>
>R077    wix/Lang/OpenJDK.Base.de-de.wxl wix/Lang/OpenJDK.Base.de-de.wxl.template
>R077    wix/Lang/OpenJDK.Base.en-us.wxl wix/Lang/OpenJDK.Base.en-us.wxl.template
>R077    wix/Lang/OpenJDK.Base.es-es.wxl wix/Lang/OpenJDK.Base.es-es.wxl.template
>R077    wix/Lang/OpenJDK.Base.fr-fr.wxl wix/Lang/OpenJDK.Base.fr-fr.wxl.template
>R077    wix/Lang/OpenJDK.Base.ja-jp.wxl wix/Lang/OpenJDK.Base.ja-jp.wxl.template
>R076    wix/Lang/OpenJDK.Base.zh-cn.wxl wix/Lang/OpenJDK.Base.zh-cn.wxl.template
>R076    wix/Lang/OpenJDK.Base.zh-tw.wxl wix/Lang/OpenJDK.Base.zh-tw.wxl.template

**Comment by junyuan-z** [comment]
@aahlenst @douph1 ready to be reviewed again.

**Comment by douph1** [comment]
Some strange templating replacement are running accross jdk https://github.com/AdoptOpenJDK/openjdk-installer/pull/252/checks?check_run_id=1732121741
`string replacement input D:\a\openjdk-installer\openjdk-installer\wix\SourceDir\OpenJDK8\hotspot\x64\jdk8u265-b01\jre\lib\management\jmxremote.password.template output D:\a\openjdk-installer\openjdk-installer\wix\SourceDir\OpenJDK8\hotspot\x64\jdk8u265-b01\jre\lib\management\jmxremote.password
string replacement input D:\a\openjdk-installer\openjdk-installer\wix\SourceDir\OpenJDK8\hotspot\x64\jdk8u265-b01\jre\lib\management\snmp.acl.template output D:\a\openjdk-installer\openjdk-installer\wix\SourceDir\OpenJDK8\hotspot\x64\jdk8u265-b01\jre\lib\management\snmp.acl
string replacement input D:\a\openjdk-installer\openjdk-installer\wix\SourceDir\OpenJDK8\hotspot\x64\jdk8u265-b01-jre\lib\management\jmxremote.password.template output D:\a\openjdk-installer\openjdk-installer\wix\SourceDir\OpenJDK8\hotspot\x64\jdk8u265-b01-jre\lib\management\jmxremote.password
string replacement input D:\a\openjdk-installer\openjdk-installer\wix\SourceDir\OpenJDK8\hotspot\x64\jdk8u265-b01-jre\lib\management\snmp.acl.template output D:\a\openjdk-installer\openjdk-installer\wix\SourceDir\OpenJDK8\hotspot\x64\jdk8u265-b01-jre\lib\management\snmp.acl`

**Review by douph1 — CHANGES_REQUESTED** [review]
See previous comment about search/replace in *.template recursively inside the jdk source

**Review by douph1 — APPROVED** [review]

**Comment by karianna** [comment]
@junyuan-z - this needs a rebase and then we can get final approval from @aahlenst

**Comment by karianna** [comment]
@aahlenst Can you re-review this?

File: wix/Lang/OpenJDK.dragonwell.de-de.wxl.template [hunk 13]
```diff
@@ -0,0 +1,12 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (aahlenst) [inline wix/Lang/OpenJDK.dragonwell.de-de.wxl.template:1]: This whole file makes no sense to me. We have sentences in here like "AdoptOpenJDK Runtime Environment with Alibaba Dragonwell" and "AdoptOpenJDK JRE with Alibaba Dragonwell" and "JRE with Dragonwell". What's the naming concept again?

Comment (aahlenst) [reply wix/Lang/OpenJDK.dragonwell.de-de.wxl.template:1]: The English version has the same problem.

**Review by aahlenst — APPROVED** [review]
The product naming got confusing before this patch was introduced, so I let it slide. We can clean it up when it's time for Adoptium.

**Comment by karianna** [comment]
> The product naming got confusing before this patch was introduced, so I let it slide. We can clean it up when it's time for Adoptium.

Sorry, I'm not clear - you're OK with landing this and fixing the strange German naming like AdoptOpenJDK Runtime Environment with Alibaba Dragonwell in a later PR?

**Pull request merged**